api/deviceadm/service.go

```diff
@@ -52,15 +52,43 @@ func (s *service) GetDevice(ctx context.Context, uid models.UID) (*models.Device
 	return s.store.GetDevice(ctx, uid)
 }
 
-func (s *service) DeleteDevice(ctx context.Context, uid models.UID, tenant string) error {
+func (s *service) DeleteDevice(ctx context.Context, uid models.UID, tenant, username string) error {
+	user, err := s.store.GetUserByUsername(ctx, username)
+	if err != nil {
+		return err
+	}
+
+	namespace, err := s.store.GetNamespace(ctx, tenant)
+	if err != nil {
+		return err
+	}
+
+	if user.ID != namespace.Owner {
+		return ErrUnauthorized
+	}
```

Comment:
This is being repeated, so it's better to rework this to be a generic function called multiple times. Something like: `IsNamespaceOnwer(tenant, username) bool` or similar.